### PR TITLE
feat: adds no-global-fetch rule to avoid direct usage of fetch in favour of our thin wrapper

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

File: plugins/eslint-plugin-liferay-portal/lib/rules/no-global-fetch.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		fixable: null,
+		messages: {
+			noGlobalFetch: DESCRIPTION,
+		},
+		schema: [],
+		type: 'problem',
+	},
+
+	create(context) {
+		let foundFetchImport = false;
```
Comment (jbalsas) [inline plugins/eslint-plugin-liferay-portal/lib/rules/no-global-fetch.js:27]: I think this is sufficiently scoped per-context... if it's not, we can store this using some additional `context` information.

File: README.md [hunk 2]
```diff
@@ -66,6 +66,7 @@ The bundled `eslint-plugin-liferay` plugin includes the following [rules](./plug
 
 The bundled `eslint-plugin-liferay-portal` plugin includes the following [rules](./plugins/eslint-plugin-liferay-portal/docs/rules):
 
+-   [liferay-portal/no-global-fetch](./plugins/eslint-plugin-liferay-portal/docs/rules/no-global-fetch.md): Prevents usage of unwrapped fetch to avoid possible issues related to security misconfiguration.
```
Comment (wincent) [inline README.md:69]: Yeah, obviously we need a Markdown linter to enforce order here. 😂

File: plugins/eslint-plugin-liferay-portal/docs/rules/no-global-fetch.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Disallow use of global fetch (no-global-fetch)
+
+This rule guards against the direct use of the `fetch` API. As a secured-environment, Liferay Portal requests often rely on specific security headers and tokens being set in the requests. The `frontend-js-web` module offers a thin wrapp around `fetch` that takes care
```
Comment (wincent) [inline plugins/eslint-plugin-liferay-portal/docs/rules/no-global-fetch.md]: wrapp → wrapper

File: plugins/eslint-plugin-liferay-portal/docs/rules/no-global-fetch.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+of the most common configuration to avoid issues.
```
Comment (wincent) [inline plugins/eslint-plugin-liferay-portal/docs/rules/no-global-fetch.md]: No need to break this one onto a new line.

File: plugins/eslint-plugin-liferay-portal/index.js [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 module.exports = {
 	rules: {
+		'no-global-fetch': require('./lib/rules/no-global-fetch'),
```
Comment (wincent) [inline plugins/eslint-plugin-liferay-portal/index.js]: Should order this too, although I bet it will conflict with your other PR.

**Review by wincent — APPROVED** [review]
lg2m!

**Comment by jbalsas** [comment]
Ok, let me rebase and try to format things properly :D

**Comment by jbalsas** [comment]
Done, @wincent! I've rebased and pushforced with an additional commit for the formatting issues.

**Pull request merged**